content/blog/monthly-july-19.md

```diff
@@ -0,0 +1,128 @@
+* Source code reading posts on [an analysis of control flow through service layers](https://pingcap.com/blog-cn/tikv-source-code-reading-9/), [Snapshot send and receive](https://pingcap.com/blog-cn/tikv-source-code-reading-10/), and [Storage and transaction control](https://pingcap.com/blog-cn/tikv-soucre-code-reading-11/) (Chinese).
+
+
+## Current projects
+
+There's lots going on in the TiKV ecosystem, here we'll try to highlight some of the interesting projects.
+
+### Performance
+
+There's ongoing work to measure and improve TiKV's performance in all areas. We hope to report more on specifics soon. We've been making heavy use of tools like Intel Vtune and flamegraphs to find the slowest parts of TiKV. We hope to report more on specifics soon.
+
+
+### Titan
+
+[Titan](https://github.com/pingcap/titan) is a storage engine which improves on RocksDB. It is part of TiKV already and you can try it out by following [these instructions](https://tikv.org/docs/3.0/tasks/configure/titan/). Note that you can't go back to RocksDB after enabling Titan. We're working to improve its performance and stability. You can read more in this [blog post](https://pingcap.com/blog/titan-storage-engine-design-and-implementation/).
+
+
+### Docs
+
+We are currently migrating [our documentation](https://tikv.org/docs/3.0/concepts/overview/) from the TiKV's wiki to the official website. We'll also be refactoring much of our documentation over the coming months. If you feel something is lacking please let us know!
+
+
+### Replica read
+
+Replica read allows a database to read from a Raft follower, not just the leader. This greatly improves read performance when the client is close to a follower, but far from the leader (e.g., in different data centres). This feature is implemented in TiKV and we are working on adding support to TiDB.
+
+
+### Protobuf implementation
+
+We're experimenting with different implementations of Protocol Buffers (which underpin all our RPC communication). We've added support for [Prost](https://github.com/danburkert/prost) to our supporting libraries, and are working on adding support to TiKV.
+
+
+### Testing and stability
+
+We're working on making testing more reliable, adding long-running tests to our CI, and monitoring performance.
+
+
+### Adding RPN functions
+
+TiKV supports executing SQL functions on the database using its coprocessor. We've been working to support RPN (reverse polish notation) functions, which are an optimisation technique for coprocessor. In the past month we've added support for `coalesce`, `in`, `case_when`, `if`, `abs`, `multiply`, `divide`, and `date_format`. There is [more information](https://pingcap.com/blog/adding-built-in-functions-to-tikv/) about implementing coprocessor functions.
+
+
+### Rust client
+
+We're working on implementing a [client library](https://github.com/tikv/client-rust/) for TiKV in Rust. This will support very high performance TiKV clients.
```

Comment:
```suggestion
We're working on implementing a [client library](https://github.com/tikv/client-rust/) for TiKV in Rust. This will support TiKV clients with very high performance.
```

Reply:
I think the original sounds more natural

Reply:
Well, "very high performance" is a noun with an adverb and adjective as its modifiers already. So using them as the modifier for another noun (TiKV clients) does not look a balanced and natural structure to me.  This is more like a preference, not a matter of right or wrong :)  I am fine either way.

Reply:
I think in this case "high performance" is used as an adjective (it should probably be "high-performance", but that looks a little bit odd to me). I can't explain why the other version feels worse, maybe the "with" feels a little bit out of place?